controllers/etcd_controller.go

```diff
@@ -929,6 +946,44 @@ func (r *EtcdReconciler) getCronJobFromEtcd(etcd *druidv1alpha1.Etcd, values map
 	return nil, fmt.Errorf("missing cronjob template file in the charts: %v", cronJobPath)
 }
 
+func (r *EtcdReconciler) getCopyJobFromEtcd(etcd *druidv1alpha1.Etcd, values map[string]interface{}, logger logr.Logger) (*batchv1.Job, error) {
+	decoded := &batchv1.Job{}
+	copyJobPath := getChartPathForCopyJob()
+	chartPath := getChartPath()
+
+	renderedChart, err := r.chartApplier.Render(chartPath, etcd.Name, etcd.Namespace, values)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := decodeObject(renderedChart, copyJobPath, &decoded); err != nil {
+		return nil, err
+	}
+
+	return decoded, nil
+}
+
+func (r *EtcdReconciler) runCopyJob(ctx context.Context, etcd *druidv1alpha1.Etcd, values map[string]interface{}, logger logr.Logger) error {
+	job, err := r.getCopyJobFromEtcd(etcd, values, logger)
+	if err != nil {
+		return err
+	}
+	if err := r.Get(ctx, kutil.Key(job.Namespace, job.Name), &batchv1.Job{}); err == nil {
+		logger.Info(fmt.Sprintf("Job %s/%s is already running", job.Namespace, job.Name))
+		return nil
+	}
+
+	return r.Create(ctx, job)
+}
+
+func decodeObject(renderedChart *chartrenderer.RenderedChart, path string, object interface{}) error {
```

Comment:
Since we have introduced this function, how about using it also in other places with similar logic, e.g. line 938 above.